### PR TITLE
Ocp on aws storage api

### PR DESCRIPTION
**Pull request by adberglund**
## Summary
This creates a storage endpoint for OCP which currently uses OCP on AWS data. 

## Examples

To test I generate nise data for OCP and AWS that share a resource_id. Then update the reporting_ocpawscostlineitem_daily_summary table to have product_family = 'Storage' for a single line item that I can easily track.

A note: The units down below show "Hrs" because I was lazy and didn't also update the units column of the nise data I modified. So I should have also set units = 'GB-Mo' 

`GET /local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_units]=month&filter[time_scope_value]=-2&filter[resolution]=monthly&group_by[tag:environment]=*`

```
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "group_by": {
        "tag:environment": [
            "*"
        ]
    },
    "filter": {
        "resolution": "monthly",
        "time_scope_value": "-2",
        "time_scope_units": "month"
    },
    "data": [
        {
            "date": "2019-01",
            "environments": [
                {
                    "environment": "dev",
                    "values": [
                        {
                            "environment": "dev",
                            "date": "2019-01",
                            "units": "Hrs",
                            "cost": 24.0,
                            "total": 24.0
                        }
                    ]
                }
            ]
        }
    ],
    "total": {
        "units": "Hrs",
        "cost": 24.0,
        "total": 24.0
    }
}
```

`http://127.0.0.1:8000/local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_units]=month&filter[time_scope_value]=-2&filter[resolution]=monthly&group_by[project]=*`

```
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "group_by": {
        "project": [
            "*"
        ]
    },
    "filter": {
        "resolution": "monthly",
        "time_scope_value": "-2",
        "time_scope_units": "month"
    },
    "data": [
        {
            "date": "2019-01",
            "projects": [
                {
                    "project": "namespace_ci",
                    "values": [
                        {
                            "date": "2019-01",
                            "units": "Hrs",
                            "project": "namespace_ci",
                            "cost": 11.674377,
                            "total": 24.0
                        }
                    ]
                }
            ]
        }
    ],
    "total": {
        "units": "Hrs",
        "cost": 11.674377,
        "total": 24.0
    }
}
```

`GET /local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_units]=month&filter[time_scope_value]=-2&filter[resolution]=monthly&group_by[node]=*`

```
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "group_by": {
        "node": [
            "*"
        ]
    },
    "filter": {
        "resolution": "monthly",
        "time_scope_value": "-2",
        "time_scope_units": "month"
    },
    "data": [
        {
            "date": "2019-01",
            "nodes": [
                {
                    "node": "alpha",
                    "values": [
                        {
                            "node": "alpha",
                            "date": "2019-01",
                            "units": "Hrs",
                            "cost": 24.0,
                            "total": 24.0
                        }
                    ]
                }
            ]
        }
    ],
    "total": {
        "units": "Hrs",
        "cost": 24.0,
        "total": 24.0
    }
}
```

`http://127.0.0.1:8000/local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_units]=month&filter[time_scope_value]=-2&filter[resolution]=monthly&group_by[account]=*`

```
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "group_by": {
        "account": [
            "*"
        ]
    },
    "filter": {
        "resolution": "monthly",
        "time_scope_value": "-2",
        "time_scope_units": "month"
    },
    "data": [
        {
            "date": "2019-01",
            "accounts": [
                {
                    "account": "9999999999999",
                    "values": [
                        {
                            "date": "2019-01",
                            "units": "Hrs",
                            "account": "9999999999999",
                            "cost": 24.0,
                            "total": 24.0,
                            "account_alias": "9999999999999"
                        }
                    ]
                }
            ]
        }
    ],
    "total": {
        "units": "Hrs",
        "cost": 24.0,
        "total": 24.0
    }
}
```

### CODE REVIEW

File: koku/api/report/aws/aws_query_handler.py [hunk 1]
```diff
@@ -51,10 +49,11 @@ def __init__(self, query_parameters, url_data,
             tenant    (String): the tenant to use to access CUR data
             kwargs    (Dict): A dictionary for internal query alteration based on path
         """
-        kwargs['provider'] = 'AWS'
-        kwargs['no_tag_query'] = QueryFilter(operation='tags__iexact', parameter='{}')
+        if not kwargs.get('provider'):
```
Comment (dccurtis) [inline koku/api/report/aws/aws_query_handler.py]: What's this check for?

Comment (adberglund) [reply koku/api/report/aws/aws_query_handler.py]: The `OCPAWSReportQueryHandler` is a subclass of the `AWSReportQueryHandler` and when it super inits, the provider was getting reset to AWS

Comment (dccurtis) [reply koku/api/report/aws/aws_query_handler.py]: ah I see, thanks for the explanation.

File: koku/api/report/ocp_aws/ocp_aws_query_handler.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                         tenant, **kwargs)
+        self.update_cost_field()
+
+    def update_cost_field(self):
+        """Update which field is used to calculate cost by group by param."""
+        group_by = self._get_group_by()
+        self._mapper = copy.deepcopy(self._mapper)
+        if group_by and group_by[0] == 'project':
+            self._mapper._report_type_map['aggregates']['cost'] = Sum('pod_cost')
+            self._mapper._report_type_map['annotations']['cost'] = Sum('pod_cost')
```
Comment (blentz) [inline koku/api/report/ocp_aws/ocp_aws_query_handler.py]: You shouldn't be updating the map in the code. If the map isn't right, fix the map.

The whole point of having it is to separate the data fields from the code. So, it should be considered read-only.

Comment (adberglund) [reply koku/api/report/ocp_aws/ocp_aws_query_handler.py]: project cost is a special case where we only want to the sum of the pod specific costs to come through. Other times we want to just sum the AWS costs (for node, or cluster, or account, et cetera...) So we'll have to at least update what part of the map we're pulling from dynamically based on the group by argument of this API.

Comment (blentz) [reply koku/api/report/ocp_aws/ocp_aws_query_handler.py]: Yep. That can be two keys in a dict where you conditionally select which key to use.

It's just the reverse of what you have. :slightly_smiling_face:

**Pull request merged**